apps/server/src/modules/tool/external-tool/controller/api-test/tool-configuration.api.spec.ts

```diff
@@ -122,10 +123,13 @@ describe('ToolConfigurationController (API)', () => {
 
 				const course: Course = courseFactory.buildWithId({ teachers: [teacherUser], school });
 
+				const baseUrl = Configuration.get('PUBLIC_BACKEND_URL') as string;
 				const [globalParameter, schoolParameter, contextParameter] = customParameterFactory.buildListWithEachType();
 				const externalTool: ExternalToolEntity = externalToolEntityFactory.buildWithId({
+					logoBase64: 'logo',
 					parameters: [globalParameter, schoolParameter, contextParameter],
 				});
+				externalTool.logoUrl = `${baseUrl}/v3/tools/external-tools/${externalTool.id}/logo`;
```

Comment:
Why in a new line?

Reply:
because in that line i need the id of the tool, which is created by the factory

Reply:
Okay...
Hint: You could create a own ObjectId. Call the factory only with build() an then you use the id.